content/docs/reference/design-docs/train-sim-v3/_index.en.md

```diff
@@ -0,0 +1,514 @@
+- when a transition between two margin values occurs, slow downs occur before value changes, and speed ups
+  after value changes. This is nice in theory, because it makes the graphs look nicer. The downside is that it
+  makes margin values interdependent at each slow-down, as how much speed needs to be lost affects the time lost
+  in the section.
+
+
+### Input modeling
+
+With the previous implementation, the simulation takes sequence of constraint position and speed curves as an input
+(continuous in position, can be discontinuous in speed), and produces a continuous curve.
+
+The output is fine, but the input is troublesome:
+
+- braking curves have to be part of constraint curves
+- these constraint curves don't have a direct match with actual constraints, such as speed limits, stops, or reaction to signal
+- constraints cannot evolve over time, and cannot be interpreted differently depending on when the train reached these constraints
+- constraints cannot overlap. the input is pre-processed to filter out obscured constraints
+
+## Design specification
+
+```mermaid
+flowchart TD
+subgraph Input
+    InitTrainState[initial train state]
+    PathPhysicsProps[path physics properties]
+    AbstractDrivingInstructions[abstract driving instructions]
+    TargetSchedule[target schedule]
+end
+
+DrivingInstructionCompiler([driving instruction compiler])
+ConcreteDrivingInstructions[driving instructions + limits]
+ScheduleController([schedule controller])
+DriverBehaviorModule([driver behavior module])
+
+TargetSchedule --> ScheduleController
+ScheduleController -- adjusts slowdown coefficient --> DriverBehaviorModule
+AbstractDrivingInstructions --> DrivingInstructionCompiler
+PathPhysicsProps --> DrivingInstructionCompiler
+ScheduleController -- tracks train state --> TrainSim
+
+DriverBehaviorModule -- makes decisions --> TrainSim
+ConcreteDrivingInstructions ---> DriverBehaviorModule
+DrivingInstructionCompiler ---> ConcreteDrivingInstructions
+
+InitTrainState ---> ScheduleController
+
+TrainSim --> SimResults
+
+TrainSim([train simulator])
+SimResults[simulation result curve]
+```
+
+
+### Driving instructions
+
+Driving instructions model what the train has to do, and under what conditions.
+Driving instructions are generated using domain constraints such as:
+
+- unsignaled line speed limits
+- permanent signaled speed limits
+- temporary speed limits
+- dynamic signaling:
+  - block / moving block
+  - dynamically signaled speed restrictions
+- neutral zones
+- stops
+- margins
+
+
+There are two types of driving instructions:
+
+- **Abstract driving instructions** model the high-level, rolling stock independent
+range of acceptable behavior: reach 30km/h at this location
+- **Concrete driving instructions** model the specific range of acceptable behavior
+for a specific rolling stock, using limit curves: don't go faster than this curve
+
+```mermaid
+flowchart TD
+Constraint[constraint]
```

Comment:
Constraints are introduced here but not in the upper diagram, what are their relation ?